fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/flaps_channel.rs

```diff
@@ -0,0 +1,371 @@
+use crate::systems::shared::arinc429::{Arinc429Word, SignStatus};
+use systems::hydraulic::command_sensor_unit::{CSUMonitor, CSU};
+use systems::hydraulic::flap_slat::{ChannelCommand, SolenoidStatus, ValveBlock};
+use systems::shared::{AdirsMeasurementOutputs, PositionPickoffUnit};
+
+use systems::simulation::{
+    InitContext, SimulationElement, SimulationElementVisitor, SimulatorWriter, UpdateContext,
+    VariableIdentifier, Write,
+};
+
+use uom::si::{angle::degree, f64::*, velocity::knot};
+
+use super::sfcc::SlatFlapControlComputerMisc;
+
+pub struct FlapsChannel {
+    flaps_fppu_angle_id: VariableIdentifier,
+    flap_actual_position_word_id: VariableIdentifier,
+    fap_ids: [VariableIdentifier; 7],
+
+    flaps_demanded_angle: Angle,
+    flaps_feedback_angle: Angle,
+
+    csu_monitor: CSUMonitor,
+
+    conf1_flaps: Angle,
+    conf1f_flaps: Angle,
+    kts_100: Velocity,
+    kts_210: Velocity,
+
+    // OUTPUTS
+    fap: [bool; 7],
+
+    flap_auto_command_active: bool,
+    flap_auto_command_engaged: bool,
+    flap_auto_command_angle: Angle,
+}
+
+impl FlapsChannel {
+    const FLAP_CONF1_FPPU_ANGLE: f64 = 0.; //deg
+    const FLAP_CONF1F_FPPU_ANGLE: f64 = 120.21; //deg
+    const KNOTS_100: f64 = 100.;
+    const KNOTS_210: f64 = 210.;
+
+    pub fn new(context: &mut InitContext, num: u8) -> Self {
+        Self {
+            flaps_fppu_angle_id: context.get_identifier("FLAPS_FPPU_ANGLE".to_owned()),
+            flap_actual_position_word_id: context
+                .get_identifier(format!("SFCC_{num}_FLAP_ACTUAL_POSITION_WORD")),
+            fap_ids: [
+                context.get_identifier(format!("SFCC_{num}_FAP_1")),
+                context.get_identifier(format!("SFCC_{num}_FAP_2")),
+                context.get_identifier(format!("SFCC_{num}_FAP_3")),
+                context.get_identifier(format!("SFCC_{num}_FAP_4")),
+                context.get_identifier(format!("SFCC_{num}_FAP_5")),
+                context.get_identifier(format!("SFCC_{num}_FAP_6")),
+                context.get_identifier(format!("SFCC_{num}_FAP_7")),
+            ],
```

Comment:
```suggestion
            fap_ids: [1, 2, 3, 4, 5, 6, 7]
                .map(|id| context.get_identifier(format!("SFCC_{num}_FAP_{id}"))),
```
**Reasoning**: reduce code duplication